src/internal_telemetry/allocations/allocator/tracing_allocator.rs

```diff
@@ -26,62 +26,56 @@ impl<A, T> GroupedTraceableAllocator<A, T> {
     }
 }
 
-impl<A: GlobalAlloc, T: Tracer> GroupedTraceableAllocator<A, T> {
-    #[inline(always)]
-    unsafe fn get_wrapped_allocation(&self, object_layout: Layout) -> (*mut u8, *mut u8, Layout) {
+unsafe impl<A: GlobalAlloc, T: Tracer> GlobalAlloc for GroupedTraceableAllocator<A, T> {
+    #[inline]
+    unsafe fn alloc(&self, object_layout: Layout) -> *mut u8 {
+        if !TRACK_ALLOCATIONS.load(Ordering::Relaxed) {
```

Comment:
I think I'd want this to be `Acquire`, along with the callsite where we check if allocation tracking is enabled.

This would just provide a stronger guarantee that things are ordered as we expect / need them to be for our invariants to hold.

Alternatively, maybe this is fine as `Relaxed` because when Vector starts, only the main thread exists, and so there shouldn't yet be any computation that would lead to interleaved load/stores where the ordering would matter... but I'd have to think about that one a little more because I'm not as much of an expert in the memory ordering to feel confident making that determination in this comment alone. 😅

Reply:
Hmm yeah. I am curious about the potential overhead.